src/audit/cache_poisoning.rs

```diff
@@ -0,0 +1,292 @@
+use crate::audit::{audit_meta, WorkflowAudit};
+use crate::finding::{Confidence, Finding, Severity};
+use crate::models::Step;
+use crate::state::AuditState;
+use github_actions_models::common::expr::ExplicitExpr;
+use github_actions_models::common::{Env, EnvValue};
+use github_actions_models::workflow::event::{BareEvent, OptionalBody};
+use github_actions_models::workflow::job::StepBody;
+use github_actions_models::workflow::Trigger;
+use std::ops::Deref;
+use std::sync::LazyLock;
+
+/// The possible toggle types for cache-aware Actions.
+/// A boolean toggle means only one possible literal value (true)
+/// to opt into the cache behavior, whereas a multi-value toggle means
+/// two or more values defining opt in for caching
+#[derive(PartialEq)]
+enum CacheControlToggle {
+    Boolean,
+    MultiValue,
+}
+
+/// The general schema for a cache-aware actions
+struct CacheAwareAction {
+    /// The owner/repo part within the Action full coordinate
+    reference: &'static str,
+    /// The name of the input the toggle caching behavior
+    cache_control_input_name: &'static str,
+    /// Whether this input defines opt-in behavior
+    cache_control_input_defines_opt_in: bool,
```

Comment:
Thoughts on collapsing these into an enumerated state? e.g.:

```
cache_control: CacheControl::OptIn("field-name")
cache_control: CacheControl::OptOut("field-name")
```

(and maybe `cache_control_input_toggle` too, although I haven't thought too hard on that one.)

Reply:
There is definitely some room to escape the primitives obsession here 😄

Reply:
Also improved [here](https://github.com/woodruffw/zizmor/pull/294/commits/85475ab269837f423dceab783fac81f838a20cee). Reads much better now, btw!